### PR TITLE
Replace .travis.yml with a Github Actions Workflow.

### PR DESCRIPTION
**Pull request by Dretch**
travis-ci.org is going away: all projects must migrate to travis-ci.com,
where each project gets 1000 free minutes before they have to contact
Travis to ask for more (i.e. the free minutes are a one-off, not per
month).

Github Actions offers 2000 free minutes per month, per repo.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v12
+      - uses: cachix/cachix-action@v8
+        with:
+          name: gi-gtk-declarative
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
```
Comment (Dretch) [inline .github/workflows/main.yml:14]: This won't work unless a repository secret is created in the Github repo settings, as per: https://nix.dev/tutorials/continuous-integration-github-actions.html

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      - name: 'Build and Test Libraries and Examples'
+        run: nix-build -j2 --arg doCheck true
+      - name: 'Build Docs'
+        run: nix-shell -j2 --run 'cd docs && mkdocs build'
```
Comment (Dretch) [inline .github/workflows/main.yml:18]: The travis build didn't build the docs, but I guess this is a handy thing to have.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v12
+      - uses: cachix/cachix-action@v8
+        with:
+          name: gi-gtk-declarative
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+      - name: 'Build and Test Libraries and Examples'
+        run: nix-build -j2 --arg doCheck true
```
Comment (Dretch) [inline .github/workflows/main.yml:16]: The travis build did `nix-build -j2 -A gi-gtk-declarative --arg doCheck true` and then separately did `nix-build -j2 -A examples`, but I don't understand why (possibly it is related to Cachix), so this just builds everything together.

**Comment by Dretch** [comment]
If you want to see (an approximation of) what this will look like once it has been merged, see here: https://github.com/Dretch/gi-gtk-declarative/actions?query=branch%3Acomponents

**Comment by owickstrom** [comment]
Awesome work @Dretch !

**Review by owickstrom — APPROVED** [review]

**Comment by owickstrom** [comment]
I'll add the secret for Cachix.

**Pull request merged**